app/logic/courseManagement.py

```diff
@@ -41,3 +43,25 @@ def getinstructorData(courseIds):
         instructorDict.setdefault(i.course.id, []).append(i.user.firstName + " " + i.user.lastName)
 
     return instructorDict
+
+def updateCourse(courseData, instructorsDict):
+    for toggler in ["regularOccurenceToggle", "slSectionsToggle", "permanentDesignation"]:
+        courseData.setdefault(toggler, "off")
+    status = CourseStatus.get(CourseStatus.status == "Pending")
+    course = Course.update(
+        courseName=courseData["courseName"],
+        courseAbbreviation=courseData["courseAbbreviation"],
+        courseCredit=courseData["credit"],
+        isRegularlyOccuring=1 if "on" in courseData["regularOccurenceToggle"] else 0,
+        term=courseData['term'],
+        status=status,
+        isAllSectionsServiceLearning=1 if "on" in courseData["slSectionsToggle"] else 0,
+        serviceLearningDesignatedSections=courseData["slDesignation"],
+        isPermanentlyDesignated=1 if "on" in courseData["permanentDesignation"] else 0,

```

Comment:
More of a preference thing, but these boolean fields could be assigned with something a little shorter that uses booleans and avoids if/else, like: 
`isPermanentlyDesignated=("on" in courseData["permanentDesignation"])